src/files.ts

```diff
@@ -71,6 +73,7 @@ export interface FileX {
 
 export function getFileMethods(
     linkBuilder: (path: string) => string | URL,
+    pathBuilder: (path: string) => string | URL | undefined,
```

Comment:
You can just pass a no-op function so you don't have to handle the `undefined` case.